src/cryptonote_core/blockchain.cpp

```diff
@@ -2800,6 +2800,19 @@ bool Blockchain::check_tx_inputs(transaction& tx, tx_verification_context &tvc,
 
   const uint8_t hf_version = m_hardfork->get_current_version();
 
+  if (hf_version >= HF_VERSION_MIN_2_OUTPUTS)
+  {
+    if (tx.version >= 2)
```

Comment:
Is it possible for this check to fail?

Reply:
Probably. Unless you find a rejection before that, but I think that's the first one. Why ?